aws/src/main/java/org/apache/iceberg/aws/dynamodb/DynamoDbCatalog.java

```diff
@@ -284,6 +284,15 @@ public boolean dropNamespace(Namespace namespace) throws NamespaceNotEmptyExcept
     }
   }
 
+  @Override
+  public boolean dropNamespace(Namespace namespace, boolean cascade)
+      throws NamespaceNotEmptyException {
+    if (cascade) {
+      listTables(namespace).forEach(this::dropTable);
```

Comment:
I think i'm missing something here since I'm not sure how dropTable get's called here with a single arg. The issue like I mentioned above is this would be a purely metadata delete for Dynamo I think in this case. Is that the expected behavior we want?

Reply:
dropTable is overloaded and has two methods, 1 with just single arg TableIdentifier and another which takes the `boolean purge` arg, we are using the former one here where purge is false.

addressed the purge comment here (TBD)